fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FunctionBinder.java

```diff
@@ -188,6 +191,15 @@ public Expression visitUnboundFunction(UnboundFunction unboundFunction, Expressi
                 // so wrap COUNT with Nvl to ensure it's result is 0 instead of null to get the correct result
                 boundFunction = new Nvl(boundFunction, new BigIntLiteral(0));
             }
+            if (context.cascadesContext.currentElementAtLevel() == 1
+                        && PushDownToProjectionFunction.validToPushDown(boundFunction)) {
+                // Only rewrite the top level of PushDownToProjectionFunction, otherwise invalid slot will be generated
+                context.cascadesContext.resetCurrentElementAtLevel();
```

Comment:
does it could work well for expression such as `element_at(v1, cast(element_at(v2, 'a') as int) + cast(element_at(v3, 'b') as bigint))`?

Reply:
this function can not be pushed down now， so this pr has no effect for such expression